pilot/pkg/model/service.go

```diff
@@ -541,6 +541,16 @@ type ServiceDiscovery interface {
 
 	// NetworkManager returns a list of network gateways that can be used to access endpoints residing in this registry..
 	NetworkGateways() []*NetworkGateway
+
+	// ExportedServices returns a list of services exported by cluster.
+	ExportedServices() []ExportedService
```

Comment:
It seems a bit heavy to add an interface to ServiceDiscovery, which is (and will be forever?) only implemented by a single controller and only for debug interface. Maybe we can just directly access the registry if it exists or something?

Reply:
I don't disagree that it's a little heavy handed. However, it is just following the pattern established with gateways. I suppose it's possible that other types of clusters could provide some sort of MCS-like mechanism.

In general, I think I prefer this to having some sort of break-glass mechanism that directly accesses kube controllers. IDK ... WDYT?

@stevenctl any opinion?

Reply:
I sort-of lean towards John's side here. I think if it's for debug purposes we can just do some casts from `ServiceDiscovery -> Aggregate` then call `GetRegistries` and cast to k8s controllers etc etc. 

If we were using this for actual generation logic I would probably lean the other way.

Reply:
Or even implementing it in aggregate cluster and just leaving  it off the interface